### ISSUE TITLE
Firmware update fails to run on Thinkpad z13

### ISSUE THREAD

**Issue by kcroker** [fwupd/firmware-lenovo#352]
I was having issues using devuan fwupdmgr to install the BIOS updates for my laptop.  So I spent some time upgrading fwupd from debian unstable and fwupd-efi from github, as some of the issues logs seemed to show that old versions were problematic with newer firmware updates.

I'm now using fwupd 1.9.1 and just built an fwupdx64.efi following the README.md.  I positioned it in the right place for my distribution to install it when running upgrade-grub.  I run fwupdmgr and select the first update.  That works, but when I reboot, the firmware update still fails to run.  

An error message flashes briefly, and I can see (when making a video of the boot and freezeframing with my phone):

```
fwupd-efi version 1.5
WARNING: Could not update variable status for 'fwupd- <lots of hex>': Out of Resources
WARNING: Could not set update status: Out of Resources
```

I have 186MB available on my EFI partition.  The new firmware cab are under 20MB.

Did I miss any steps here?
How can I provide diagnostic information to help diagnose and correct this issue?

**Comment by superm1** [comment]
I wonder if this is actually a gnu-efi bug.  Can you snag the `fwupdx64.efi` from the latest Fedora RPM (which should be the latest version and @hughsie very recently tested) and replace it in your filesystem in `/usr/lib/fwupd/efi` and retry it?

**Comment by kcroker** [comment]
Grabbed it from:
https://koji.fedoraproject.org/koji/buildinfo?buildID=2142105

And gave it a shot (placed it directly into the right place under `/boot/efi` )
It ran the Fedora version, because it showed v1.4 instead of v1.5.
Still no dice, looks like the same error message, though I couldn't video the error with my phone at the moment.

Not sure that it matters, but I'm running a 6.3.2 kernel.  Later 6.3.x and 6.4 kernel have real issues after waking from sleep on the z13.

**Comment by superm1** [comment]
That's very bizarre then.  This sounds like a Lenovo firmware bug, but I don't really understand how.  I just updated a Z13 myself to 1.57 quite recently.

**Comment by superm1** [comment]
> Later 6.3.x and 6.4 kernel have real issues after waking from sleep on the z13.

JFYI - I'm using 6.3.8 on a Z13 and have no problems with this.  But again I'm on the latest firmware.

**Comment by hughsie** [comment]
> Out of Resources

Could the EFI NVRAM be full with crash dumps or something?

**Comment by kcroker** [comment]
@superm1 That's very encouraging, at least I know I can get things working! xD
@hughsie Let me try to figure out what that would look like.  I'm a bit scared of bricking my device by just deleting entries in the EFI NVRAM >_<

**Comment by kcroker** [comment]
@hughsie That was indeed the issue, tons of dump files in efivars.  Removed them, things worked fine with the v1.5 fwupd-efi built natively from the yesterday's git clone.  Thanks guys, really appreciate the responsive feedback!  I also disabled the pstore logging in the kernel options, so hopefully this won't happen again.  

If you're receptive, maybe add a hint message to the error output (e.g. "efivars full of dump files") to future versions?
